thinc/backends/ops.py

```diff
@@ -878,39 +878,41 @@ def backprop_gelu(
             return dY
         return dY * dX
 
-    def mish(self, X: Floats2d, threshold: float = 20.0) -> Floats2d:
-        Y = self.alloc2f(*X.shape, dtype=X.dtype)
+    def mish(
+        self, X: FloatsType, threshold: float = 20.0, inplace: bool = False
+    ) -> FloatsType:
         tmp = X * self.xp.tanh(self.xp.log(1.0 + self.xp.exp(X)))
-        for i in range(X.shape[0]):
-            for j in range(X.shape[1]):
-                if X[i, j] >= threshold:
-                    Y[i, j] = X[i, j]
-                else:
-                    Y[i, j] = tmp[i, j]
-        return Y
+        Y = self.xp.where(X >= threshold, X, tmp)
+        if inplace:
+            X[:] = Y
+            return X
+        else:
+            return Y
 
     def backprop_mish(
         self,
-        dY: Floats2d,
+        dY: FloatsType,
         X: Floats2d,
         threshold: float = 20.0,
-        out: Optional[Floats2d] = None,
-    ) -> Floats2d:
+        inplace: bool = False,
+    ) -> FloatsType:
         xp = get_array_module(X)
         indices = X < threshold
         Xsub = X[indices]
         dYsub = dY[indices]
         omega = 4.0 * (Xsub + 1.0)
         omega += 4.0 * xp.exp(2.0 * Xsub)
+        omega += xp.exp(3.0 * Xsub)
         omega += xp.exp(Xsub) * ((4.0 * Xsub) + 6.0)
         delta = 2.0 * xp.exp(Xsub)
         delta += xp.exp(2.0 * Xsub)
         delta += 2.0
```

Comment:
Not sure if its faster or not, but to me it seems like `delta` could be written shorter. 

Like at the moment its `2 * e^x + e^{2 * x} + 2`.

This expression seems to me like `1 + (e^x + 1)^2` . Like if we do the binomial expansion we get `1 + e^x^2 + 2 * e^x * 1 + 1^2` , which is `1 + e^{2 *x} + 2 * e^x + 1`, which can be re-arranged to `2 * e^x + e^{2 *x} + 2`. I think this is correct :D

Reply:
Added.